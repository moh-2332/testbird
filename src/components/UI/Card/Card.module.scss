.card {

    &:hover {
        background-color: var(--secondary-color);
    }
}