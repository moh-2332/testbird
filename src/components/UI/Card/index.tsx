import React from 'react';

import classes from "./Card.module.scss"

const Card: React.FC<{ children: any }> = ({ children }) => {
    return (
        <div className={classes.card}>
            {children}
        </div>
    );
}

export default Card;