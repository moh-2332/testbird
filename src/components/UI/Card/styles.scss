@import '../../../styles/colors';

.card {
    &:hover {
        background-color: $secondary;
    }
}