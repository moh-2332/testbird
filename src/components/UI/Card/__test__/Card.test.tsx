import React from "react";

import { screen, render } from "@testing-library/react";
import renderer from "react-test-renderer";
import "@testing-library/jest-dom";

import Card from "../";

describe("Card tests", () => {
    it("Should match the snapshot", () => {
        const card = renderer.create(<Card><h1>This is a test</h1></Card>);
        expect(card).toMatchSnapshot();
    });

    it("Should renders correctly", () => {
        render(<Card><h1>This is a test</h1></Card>);

        const sampleText = screen.getByText(/This is a test/i);
        expect(sampleText).toBeInTheDocument();
    });
});