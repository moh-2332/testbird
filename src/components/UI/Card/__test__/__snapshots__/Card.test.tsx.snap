// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card tests Should match the snapshot 1`] = `
<div
  className="card"
>
  <h1>
    This is a test
  </h1>
</div>
`;
